cmd/rdma/main.go

```diff
@@ -136,16 +175,69 @@ func (plugin *rdmaCniPlugin) moveRdmaDevFromNs(rdmaDev string, nsPath string) er
 	}
 	defer targetNs.Close()
 
+	tmpName := cRdmaDev
+	if renameReq {
+		tmpName = utils.GetRandomRdmaDevName()
+	}
 	err = sourceNs.Do(func(_ ns.NetNS) error {
-		// Move RDMA device to default namespace
-		return plugin.rdmaManager.MoveRdmaDevToNs(rdmaDev, targetNs)
+		if renameReq {
+			// Move RDMA device to default namespace
+			var sourceNsError error
+			sourceNsError = plugin.rdmaManager.SetRdmaDevName(cRdmaDev, tmpName)
+			if sourceNsError != nil {
+				log.Warn().Msgf("Failed to restore RDMA device name to its original value. %v", sourceNsError)
+				return plugin.rdmaManager.MoveRdmaDevToNs(cRdmaDev, targetNs)
+			}
+		}
+		return plugin.rdmaManager.MoveRdmaDevToNs(tmpName, targetNs)
 	})
 	if err != nil {
-		return fmt.Errorf("failed to move RDMA device %s to default namespace. %v", rdmaDev, err)
+		return fmt.Errorf("failed to move RDMA device %s to default namespace. %v", cRdmaDev, err)
+	}
+	if renameReq {
+		// set target name
+		err = targetNs.Do(func(_ ns.NetNS) error {
+			return plugin.rdmaManager.SetRdmaDevName(tmpName, sRdmaDev)
+		})
 	}
 	return err
 }
 
+func (plugin *rdmaCniPlugin) getContainerRdmaDeviceName(sRdmaDev string, nsPath string) string {
+	var err error
+	var sourceNs ns.NetNS
+	sourceNs, err = plugin.nsManager.GetNS(nsPath)
+	defer sourceNs.Close()
+	defer func() {
+		if err != nil {
+			log.Warn().Msgf("Failed to generate container RDMA device name, %s. Original name will be used.", err)
+		}
+	}()
+
+	var cRdmaDevs []string
+	err = sourceNs.Do(func(_ ns.NetNS) error {
+		var cErr error
+		cRdmaDevs, cErr = plugin.rdmaManager.GetRdmaDeviceList()
+		return cErr
+	})
+	if err != nil {
+		return sRdmaDev
+	}
+
+	// Expected format is <driver_name>_<idx>
+	s := strings.Split(sRdmaDev, "_")
```

Comment:
you can create on method that extract the driver and the index and use it here and getRdmaDevIndexFromName